pkg/userop/gas_test.go

```diff
@@ -1,12 +1,74 @@
 package userop
 
 import (
+	"context"
 	"math/big"
+	"net/url"
 	"testing"
 
+	"github.com/shopspring/decimal"
 	"github.com/stretchr/testify/require"
 )
 
+func TestGetGasPricesMiddleware(t *testing.T) {
+	t.Run("gas multipliers are applied correctly", func(t *testing.T) {
+		t.Skip("manual test")
+		providerUrl := "provider_url"
+
+		providerURL, err := url.Parse(providerUrl)
+		require.NoError(t, err)
+		providerRPC, err := NewEthBackend(*providerURL)
+		require.NoError(t, err)
+
+		tests := []struct {
+			name           string
+			gasMultipliers GasConfig
+		}{
+			{
+				name:           "no multipliers supplied result in 0 gas prices",
+				gasMultipliers: GasConfig{},
+			},
+			{
+				name: "1 multipliers do not change gas prices",
+				gasMultipliers: GasConfig{
+					MaxPriorityFeePerGasMultiplier: decimal.NewFromFloat(1),
+					MaxFeePerGasMultiplier:         decimal.NewFromFloat(1),
+				},
+			},
+			{
+				name: "1.5 multipliers are applied correctly",
+				gasMultipliers: GasConfig{
+					MaxPriorityFeePerGasMultiplier: decimal.NewFromFloat(1.5),
+					MaxFeePerGasMultiplier:         decimal.NewFromFloat(1.5),
+				},
+			},
+			{
+				name: "2.25 multipliers are applied correctly",
+				gasMultipliers: GasConfig{
+					MaxPriorityFeePerGasMultiplier: decimal.NewFromFloat(2.25),
+					MaxFeePerGasMultiplier:         decimal.NewFromFloat(2.25),
+				},
+			},
+		}
+
+		ctx := context.Background()
+		maxFeePerGas, maxPriorityFeePerGas, err := getGasPrices(ctx, providerRPC)
+		require.NoError(t, err)
+
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				gotMaxFeePerGas, gotMaxPriorityFeePerGas, err := getGasPricesAndApplyMultipliers(ctx, providerRPC, tt.gasMultipliers)
+				require.NoError(t, err)
+
+				calculatedMaxFeePerGas := decimal.NewFromBigInt(maxFeePerGas, 0).Mul(tt.gasMultipliers.MaxFeePerGasMultiplier).BigInt()
```

Comment:
use precalculated hardcoded value instead of using the same formula that you are tesing (use mocked provider instead of switch functions inside `getGasPricesAndApplyMultipliers `)

Reply:
e5ec81a4f20a633db534c8b55595ba7fef556112